### PR TITLE
Verify database state with retry

### PR DESCRIPTION
**Pull request by ScorpioCPH**
As mentioned in README

`Docker Compose may start Clair before Postgres which will raise an error. If this error is raised, manually execute docker start clair_clair.` 

so I add retry in verify database to avoid exiting.

### CODE REVIEW

**Comment by ScorpioCPH** [comment]
@Quentin-M, please review my PR, thanks!

File: database/pgsql/pgsql.go [hunk 1]
```diff
@@ -156,10 +156,26 @@ func openDatabase(registrableComponentConfig config.RegistrableComponentConfig)
...
+	// Verify database state with retry.
+	ch := make(chan int, 1)
+	go func() {
+		for {
+			err := pg.DB.Ping()
+			if err == nil {
+				ch <- 1
+			} else {
+				log.Error("failed to verify database state, retry after 2 seconds")
+				time.Sleep(2 * time.Second)
```
Comment (jzelinskie) [inline database/pgsql/pgsql.go:168]: this should be exponential backoff

File: database/pgsql/pgsql.go [hunk 2]
```diff
@@ -156,10 +156,26 @@ func openDatabase(registrableComponentConfig config.RegistrableComponentConfig)
 		return nil, fmt.Errorf("pgsql: could not open database: %v", err)
 	}
 
-	// Verify database state.
-	if err := pg.DB.Ping(); err != nil {
+	// Verify database state with retry.
+	ch := make(chan int, 1)
```
Comment (jzelinskie) [inline database/pgsql/pgsql.go:160]: `make(chan struct{})`

**Review by jzelinskie — CHANGES_REQUESTED** [review]
I think this is a fair idea, but I think that 60s is probably way too long and that we should be using exponential backoff. You can use `github.com/coreos/pkg/timeutil`'s ExpBackOff function for this.

**Comment by ajgreenb** [comment]
@jzelinskie I think this PR can now be closed?

**Comment by ScorpioCPH** [comment]
@ajgreenb Sure! Sorry for the late response.

**Pull request closed without merging**